### PR TITLE
Add AO_AUX_ONLY option to VACUUM command

### PR DESCRIPTION
**Pull request by AJR-VMware**
When this option is passed, instead of vacuuming main tables look up auxiliary tables for all indicated AO tables and vacuum those.

Also add regress test coverage of this behavior.

### CODE REVIEW

**Review by ashwinstar — COMMENTED** [review]
"If ao_aux_only option is given without a tablename, we only want all AO AUX tables" this behavior information or detail should be mentioned in commit message.

I was wondering using this option could provide capability to users to vacuum all tables containing valid relfroxenxid (which means user and auxiliary - heap tables which actually contain XIDs) and help ignore just AO tables. Though understand and agree based on name of the option why we implemented the current way. Just leaves no option to accomplish the outcome I mentioned - not even sure how much that's desired currently.

File: src/backend/commands/vacuum.c [hunk 1]
```diff
@@ -995,8 +1009,58 @@ expand_vacuum_rel(VacuumRelation *vrel, int options)
 				child_relid = parent_relid;
 			}
 		}
-	}
 
+		/*
+		 * If AO_AUX_ONLY option is passed, replace list of relations with the
+		 * auxiliary tables for each AO table in the list
```
Comment (huansong) [inline src/backend/commands/vacuum.c:1015]: Currently we hold a lock on the AO table when we vacuum its aux tables: https://github.com/greenplum-db/gpdb/blob/main/src/backend/commands/vacuum.c#L2440-L2447 (the lock is acquired [here](https://github.com/greenplum-db/gpdb/blob/main/src/backend/commands/vacuum.c#L2084-L2090)).  W/ the new mechanism we won't be doing that. Would that be an issue? Also when we vacuum the aux tables we pass `recursing=true` which indicates that we don't want to dispatch them separately: https://github.com/greenplum-db/gpdb/blob/main/src/backend/commands/vacuum.c#L2446-L2447 . I'm not yet clear what issue that might cause, will look further.

But if that turns out to be a problem, we might just open the table here. Another way which might be better is to simply bypassing tables that we don't want to vacuum in `vacuum_rel`. Also it'll be easier to achieve this goal mentioned by Ashwin (if that is still desired):

```
I was wondering using this option could provide capability to users to vacuum all tables containing valid relfroxenxid (which means user and auxiliary - heap tables which actually contain XIDs) and help ignore just AO tables
```

Comment (AJR-VMware) [reply src/backend/commands/vacuum.c:1015]: Per pairing conversation, will add additional isolation test to cover this.  Code review suggests current approach is OK, but some additional test coverage would be good.

File: src/test/isolation/specs/vacuum-ao-concurrent-drop.spec [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+}
+step drop_and_commit
+{
+    DROP TABLE ao_vac_conc;
+	COMMIT;
+}
+step begin_and_drop
+{
+    DROP TABLE ao_vac_conc;
+	BEGIN;
```
Comment (huansong) [inline src/test/isolation/specs/vacuum-ao-concurrent-drop.spec:28]: `BEGIN` should be prior to `DROP` right?

File: src/test/isolation/specs/vacuum-ao-concurrent-drop.spec [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+}
+step commit
+{
+	COMMIT;
+}
+
+session s2
+step vac_specified		{ VACUUM AO_AUX_ONLY ao_vac_conc; }
+step vac_all_tables		{ VACUUM AO_AUX_ONLY; }
```
Comment (huansong) [inline src/test/isolation/specs/vacuum-ao-concurrent-drop.spec:37]: Let's cover the case where AO_AUX_ONLY is not specified. Maybe let's just do that in `vac_all_tables` (so it just does `VACUUM`).

**Review by huansong — APPROVED** [review]
LGTM (pending CI check)! Nice tests. 

Also could we add an `Assert(!(params->options & VACOPT_AO_AUX_ONLY))` here: https://github.com/greenplum-db/gpdb/blob/main/src/backend/commands/vacuum.c#L2213-L2218 . Making sure we don't vacuum the aux tables twice somehow.

File: src/test/isolation/specs/vacuum-ao-concurrent-drop.spec (hunk 2 above)
Comment (AJR-VMware) [reply src/test/isolation/specs/vacuum-ao-concurrent-drop.spec:28]: Yep.  Fixing!

File: src/test/isolation/specs/vacuum-ao-concurrent-drop.spec (hunk 3 above)
Comment (AJR-VMware) [reply src/test/isolation/specs/vacuum-ao-concurrent-drop.spec:37]: I'll add another step+permutation that does it without the ao_aux_only option.  `VACUUM AO_AUX_ONLY` without specifying a table is a different code path for expansion than if the table is specified, so I think it'd be good to have coverage for all 3.

**Pull request merged**